clients/lsp-php.el

```diff
@@ -33,8 +33,17 @@
   :link '(url-link "https://github.com/felixfbecker/php-language-server")
   :group 'lsp-mode)
 
+(defcustom lsp-php-composer-dir
+  (if (eq (shell-command "composer -V") 0)
+      (replace-regexp-in-string "\n$" "" (shell-command-to-string "composer config --global home"))
+    "~/.composer")
+  "Home directory of composer."
+  :group 'lsp-php
+  :type 'string)
+
 (defcustom lsp-clients-php-server-command
-  `("php" ,(expand-file-name "~/.composer/vendor/felixfbecker/language-server/bin/php-language-server.php"))
+  `("php", (expand-file-name
```

Comment:
can you set `lsp-clients-php-server-command` to nil and move the calculation of the command in the lambda in :new-connection (on line 371 in original code)? The issue with the current approach is that we will increase the lsp-mode load time because each time you start lsp-mode it will execute the shell-command even if the user is not interested in php.

Reply:
Can you check it again, because as I see, line 371 is used for phpactor, not flexbecker's php-language-server.

Reply:
It should be here - https://github.com/emacs-lsp/lsp-mode/blob/b6c47e1f8b48bedfd57bf8b9dbef0a9232133506/clients/lsp-php.el#L54

Reply:
Hello,

Do you think this would work?

```
--- a/clients/lsp-php.el
+++ b/clients/lsp-php.el
@@ -41,9 +41,7 @@
   :group 'lsp-php
   :type 'string)

-(defcustom lsp-clients-php-server-command
-  `("php", (expand-file-name
-            (concat lsp-php-composer-dir "/vendor/felixfbecker/language-server/bin/php-language-server.php")))
+(defcustom lsp-clients-php-server-command nil
   "Install directory for php-language-server."
   :group 'lsp-php
   :type '(repeat string))
@@ -51,7 +49,11 @@
 (defun lsp-php--create-connection ()
   "Create lsp connection."
   (lsp-stdio-connection
-   (lambda () lsp-clients-php-server-command)
+   (lambda ()
+     (setq lsp-clients-php-server-command
+           `("php",
+             (expand-file-name
+              (concat lsp-php-composer-dir "/vendor/felixfbecker/language-server/bin/php-language-server.php")))))
    (lambda ()
      (if (and (cdr lsp-clients-php-server-command)
               (eq (string-match-p "php[0-9.]*\\'" (car lsp-clients-php-server-command)) 0))
--
```

Patch: [0001-draft-move-lsp-clients-php-server-command-to-lambda.patch.txt](https://github.com/emacs-lsp/lsp-mode/files/6295136/0001-draft-move-lsp-clients-php-server-command-to-lambda.patch.txt)

Reply:
can you set lsp-clients-php-server-command only if it is not already set?

Reply:
can you help me check the latest commit: https://github.com/emacs-lsp/lsp-mode/pull/2737/commits/de1791bd1974484aef797bf0cb2e7f4039b5b5de ?
Thanks